paasta_tools/firewall_update.py

```diff
@@ -95,16 +91,23 @@ def process_inotify_event(event, services_by_dependencies):
 
 
 def services_running_here(soa_dir):
-    # Generator helper that yields (service, instance, config) of both marathon and chronos tasks
-    cluster = load_system_paasta_config().get_cluster()
+    # Generator helper that yields (service, instance, framework, mac address)
+    # of both marathon and chronos tasks
+    for container in get_running_mesos_docker_containers():
+        if container['HostConfig']['NetworkMode'] != 'bridge':
+            continue
+
+        service = container['Labels'].get('paasta_service')
+        instance = container['Labels'].get('paasta_instance')
+
+        if service is None or instance is None:
+            continue
 
-    for service, instance, port in marathon_services_running_here():
-        config = load_marathon_service_config(service, instance, cluster, load_deployments=False, soa_dir=soa_dir)
-        yield service, instance, config
+        # TODO: this sucks and is untested, maybe we should add a label for framework?
+        framework = 'marathon' if container['Ports'] else 'chronos'
```

Comment:
You could "not care" if you wanted to, but the normal `get_instance_config` thingy has to look through each yaml file looking for an instance. (we assume they are unique across frameworks)